### PR TITLE
Add more logs and test to validate chain - Relates #2249

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
There were few edge cases where logging was not adequate. 

### How did I fix it?
Added more logs and tests. 

### Review checklist

* The PR solves #2249
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: modules/loader.js [hunk 1]
```diff
@@ -575,6 +575,13 @@ __private.loadBlockChain = function() {
  * @returns {setImmediateCallback} cb, err
  */
 __private.validateBlock = (blockToVerify, cb) => {
+	library.logger.info(
+		`[Loader][validateBlock] Validating block ${blockToVerify.id} at height ${
```
Comment (4miners) [inline modules/loader.js]: Consistency - `Loader->validateBlock`, globally in this PR
